src/main/java/com/tutorhub/security/SecurityService.java

```diff
@@ -0,0 +1,20 @@
+package com.tutorhub.security;
+
+import com.tutorhub.model.user.User;
+import com.tutorhub.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class SecurityService {
+  private final UserService userService;
+
+  public User getCurrentLoggedUser() {
+    Authentication currentLoggedUser =
+        SecurityContextHolder.getContext().getAuthentication();
+    return userService.findByUsername(currentLoggedUser.getName());
```

Comment:
does it work? I think you need to get principal and get his username

Reply:
Fixed